File: pinyougou-manager-web/src/main/webapp/js/service/itemCatService.js
app.service("itemCatService",function ($http) {
    //通过parentid查询分类
    this.findByParentId = parentId =>{
        return $http.get("/itemCat/findByParentId.do?parentId="+parentId);
    };
    // 保存
    this.save = (url,entity)=>{
        return $http.post(url,entity);
    };
    //查询某一个
    this.findOneById = (id) =>{
        return $http.get("/itemCat/findOneById.do?id="+id);
    };
    //删除
    this.dele = (ids) =>{
        return $http.get("/itemCat/delete.do?ids="+ids);
    };

    this.findAll= ()=>{
        return $http.get("/itemCat/findAll.do");
    }
});

File: pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/ItemCatService.java
package com.pinyougou.sellergoods.service;
import com.pinyougou.pojo.PageResult;
import com.pinyougou.pojo.Result;
import com.pinyougou.pojo.TbItemCat;

import java.util.List;

/**
 * 服务层接口
 * @author Administrator
 *
 */
public interface ItemCatService {


    /**
     * @Author: ZhangYu
     * @Date:2019-11-13
     * @Description: java学习
     * @Question:通过parentId查找
     */
    List<TbItemCat> findByParentId(Long parentId);

    void add(TbItemCat tbItemCat);

    void update(TbItemCat tbItemCat);

    TbItemCat findOneById(long id);

    void deleteByArrays(Long[] ids);

    List<TbItemCat> findAll();
}
